nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -49,8 +49,21 @@ def _apply_mask_and_get_affinity(seeds, niimg, radius, allow_overlap,
         radius += 1e-6
 
     clf = neighbors.NearestNeighbors(radius=radius)
-    A = clf.fit(mask_coords).radius_neighbors_graph(seeds)
-    A = A.tolil()
+    clf.fit(mask_coords)
+    A = []
+    for i, s in enumerate(seeds):
+        # Check seed is inside the mask
+        dist, _ = clf.kneighbors(s, 1)
+        if dist > 2:
+            raise ValueError('Seed #%i is too far' % i)
```

Comment:
Sorry, I don't see clearly why you chose 2 as a threshold. It is a distance  in mm ? What if the voxel grid is coarse ?